### PR TITLE
Fix for issue with type parameters on type alias

### PR DESCRIPTION
**Pull request by anubmat**
Fixing Issue #63

### CODE REVIEW

File: TypeScriptReact.YAML-tmLanguage [hunk 1]
```diff
@@ -46,13 +46,16 @@ repository:
      - include: '#enum-declaration'
 
   type-declaration:
-    name: meta.type.declaration.tsx
-    begin: \b(type)\b\s+([a-zA-Z_$][\w$]*)\s*=\s*
+    name: meta.type.declaration.ts
```
Comment (zhengbli) [inline TypeScriptReact.YAML-tmLanguage]: All grammar defined in TypeScriptReact need to have a top level scope of `.tsx`
This change is unneeded.

**Comment by zhengbli** [comment]
Please add tests to the PR. In the tests consider the following cases:
1. `<` and `>` used as "larger than" and "smaller than" signs
2. `<` and `>` used as opening and closing brackets for generic types

There was a discussion about these two cases before [at the Sublime repo](https://github.com/Microsoft/TypeScript-Sublime-Plugin/issues/285). And Bill has provided some test cases in his [comment](https://github.com/Microsoft/TypeScript-Sublime-Plugin/issues/285#issuecomment-126515395).

File: tests/cases/Issue63.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+^^type ^^Callback <^^string> = ^^(error: Error, value: T) => any
```
Comment (DanielRosenwasser) [inline tests/cases/Issue63.ts]: `string` isn't a valid type parameter name, so can you adjust the test?

**Comment by DanielRosenwasser** [comment]
Can you add a test with type literal syntax like so?

```ts
^type ^^Foo<^^T> = ^{
    ^foo: ^T,
    ^bar: ^number
^}
```

**Comment by zhengbli** [comment]
:+1:

**Pull request merged**